homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,180 @@
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import DOMAIN, SIGNAL_HEATAREA_DATA_UPDATED
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add Alpha2Climate entities from a config_entry."""
+
+    base_update_handler = hass.data[DOMAIN][config_entry.entry_id]["connection"]
+    await base_update_handler.async_update()
+
+    entities = []
+    for heatarea in base_update_handler.base.heatareas:
+        entities.append(Alpha2Climate(base_update_handler, heatarea))
+    async_add_entities(entities)
+
+
+# https://developers.home-assistant.io/docs/core/entity/climate/
+class Alpha2Climate(ClimateEntity):
+    """Alpha2 ClimateEntity."""
+
+    target_temperature_step = 0.2
+
+    def __init__(self, base_update_handler, data):
+        """Initialize Alpha2 ClimateEntity."""
+        self._base_update_handler = base_update_handler
+        self._data = data
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_HEATAREA_DATA_UPDATED,
+                self._handle_heatarea_data_updated,
+            )
+        )
+
+    def _handle_heatarea_data_updated(self, data):
+        """Handle updated heatarea data."""
+        if data["NR"] == self._data["NR"]:
+            self._data = data
+            self.async_schedule_update_ha_state()
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+    @property
+    def should_poll(self):
+        """Disable polling."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._data["HEATAREA_NAME"]
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return float(self._data.get("T_TARGET_MIN", 0))
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return float(self._data.get("T_TARGET_MAX", 30))
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return float(self._data.get("T_ACTUAL", 0))
+
+    @property
+    def hvac_mode(self):
+        """Return current operation mode."""
+        return HVAC_MODE_AUTO
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        return [HVAC_MODE_AUTO]
+
+    def set_hvac_mode(self, hvac_mode: str):
+        """Set new target hvac mode."""
+        return
+
+    @property
+    def hvac_action(self):
+        """Return the current running hvac operation if supported."""
+        return None
```

Comment:
Same as in super class
https://github.com/home-assistant/core/blob/24c1256c2c8bef2d39a2e483ac09689d850181a0/homeassistant/components/climate/__init__.py#L179
https://github.com/home-assistant/core/blob/24c1256c2c8bef2d39a2e483ac09689d850181a0/homeassistant/components/climate/__init__.py#L336-L342
```suggestion
```

Reply:
Removed.